src/test/java/io/vertx/redis/client/test/RedisOptionsTest.java

```diff
@@ -15,4 +19,24 @@ public void testRedisOptions() {
     assertEquals("redis://localhost:6379", new RedisOptions().getEndpoint());
     assertEquals(Collections.singletonList("redis://localhost:6379"), new RedisOptions().getEndpoints());
   }
+
+  @Test
+  public void testOptionsCopy() {
+    List<String> endpoints = new ArrayList<>(3);
+    endpoints.add("redis://localhost:123");
+    endpoints.add("redis://localhost:124");
+    endpoints.add("redis://localhost:125");
+
+    // Set values for which there is no default to ensure they are copied correctly
+    RedisOptions original = new RedisOptions()
```

Comment:
Rather than setting every value, I set values which had no default. This allows the test to distinguish between a value not being copied and a value having no default. I could set a value for every option and assert on their copy, but I'm not sure if we see value in that. Happy to adjust this for any feedback.